src/commands/mod.rs

```diff
@@ -972,6 +1057,55 @@ mod tests {
         ms.shutdown();
     }
 
+    #[test]
+    fn listaddresses() {
```

Comment:
Would be good to check bounds here for hardened derivation index, overflows and underflows.